src/server/game/Spells/Auras/SpellAuras.cpp

```diff
@@ -851,6 +854,31 @@ void Aura::Update(uint32 diff, Unit* caster)
             }
         }
     }
+    
+    if (m_heartBeatResistTimer)
+    {
+        if (m_heartBeatResistTimer < int32(diff))
```

Comment:
Why `int32` for a timer that can only ever be positive?

Reply:
I think `m_heartBeatResistTimer` can be negative  ([cause](https://github.com/TrinityCore/TrinityCore/pull/20206/files#diff-cd8b10cee034eef5cd33bc2e44f18482R861)).

Reply:
The linked logic subtracts a smaller value from a larger value. How do you expect it to become negative in the process?

Reply:
Well, I get that. Thank you.